components/core/src/compressor_frontend/InputBuffer.cpp

```diff
@@ -0,0 +1,111 @@
+// C++ libraries
+#include <memory.h>
+#include <string>
+
+// spdlog
+#include <spdlog/spdlog.h>
+
+// Project Headers
+#include "InputBuffer.hpp"
+
+using std::string;
+using std::to_string;
+
+namespace compressor_frontend {
+
+    void InputBuffer::reset () {
+        m_at_end_of_file = false;
+        m_finished_reading_file = false;
+        m_consumed_pos = 0;
+        m_bytes_read = 0;
+        m_last_read_first_half = false;
+        Buffer::reset();
+    }
+
+    bool InputBuffer::read_is_safe () {
+        if (m_finished_reading_file) {
+            return false;
+        }
+        // If the next message starts at 0, the previous character is at m_curr_storage_size - 1
+        if (m_consumed_pos == -1) {
```

Comment:
* Prefer `constant == variable`
* Instead of comparing against how about, `std::numeric_limits<decltype(m_consumed_pos)>::max()`?